src/windshaft/filters/collection.js

```diff
@@ -12,28 +18,21 @@ module.exports = Backbone.Collection.extend({
     var activeFilters = this.getActiveFilters();
     if (activeFilters.length) {
```

Comment:
@alonsogarciapablo  I tried to generate the same previous output but there is a case not as easy as can seem. Previously when we worked with layerIndex we generate filters for a given layerIndex and fill the "layer filter group" gaps. Now the process is different, we create a "layer filter group" and add all the layer filters.

For the next test https://github.com/CartoDB/cartodb.js/pull/1124/files#diff-1bff0c96a4fa609337cd94e943c2bfe5R109 is not as easy. Previously given three layers that only the first one has filters the out was `{ layers: [{ filters_layer_1}] }`. With the new approach it is `{ layers: [{ filters_layer_1}, {}, {}] }` and it is not easy to know if the latests "filter groups" can be removed. 

It is the only case, for other cases, for example when layer one and three has filter the output is the same: `{ layers: [{ filters_layer_1}, {}, {filter_layer_3}] }`